compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -446,10 +448,69 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                 }
             }
         }
+
         err.span_label(decl_span, "binding declared here but left uninitialized");
+        if show_assign_sugg {
+            self.suggest_assign_rvalue(&mut err, moved_place, &name, decl_span);
+        }
         err
     }
 
+    fn suggest_assign_rvalue(
+        &self,
+        err: &mut Diagnostic,
+        moved_place: PlaceRef<'tcx>,
+        name: &str,
+        decl_span: Span,
+    ) {
+        let ty = moved_place.ty(self.body, self.infcx.tcx).ty;
+        debug!("ty: {:?}, kind: {:?}", ty, ty.kind());
+
+        let initilize_msg = match ty.kind() {
+            ty::Array(_, n) => format!("[val; {}]", n),
+            ty::Int(_) | ty::Uint(_) => format!("0"),
+            ty::Float(_) => format!("0.0"),
+            ty::Bool => format!("false"),
+            ty::Never | ty::Error(_) => "".to_string(),
+            ty::Adt(def, _substs) => {
+                if format!("{:?}", def).starts_with("std::vec::Vec") {
+                    format!("vec![]")
+                } else if let Some(default_trait) = self.infcx.tcx.get_diagnostic_item(sym::Default) &&
+                    self.infcx.tcx.infer_ctxt().enter(|infcx| {
+                        infcx.type_implements_trait(default_trait, ty, ty::List::empty(), self.param_env).may_apply()
+                    }) {
+                    format!("Default::default()")
+                } else {
+                    format!("something")
```

Comment:
This seems quite sketchy to me from usability perspective, even if this _is_ a `MaybeIncorrect` diagnostic. I would rather we do not emit any diagnostic here at all in that case.

And in general, I would say we could just suggest `Default::default()` in all cases unconditionally. Inlining some of the suggestions is probably fine, but in some cases such as an `ty::Array` the suggestion similarly falls into the same bucket as `something` does here.

Alternatively, if you’d rather to keep things as-is still, maybe consider `todo!()` for a placeholder. This is what R-A does when generating code.

Reply:
I do think we should probably always suggest `todo!()` here, `Default::default` is often not what you want and with `todo!` it's a lot clearer that this is something the user has to modify.

Reply:
To give my 2c -- from what I've seen, 90% of `let var: Ty;` are meant to be `let var = Ty::default();` / `let var: Ty = Default::defaut();`, that's why in the original issue I've suggested to go this way.

But `todo!()` also feels nice, giving this clear "there has to be _something_ put here" vibes 🙂

Reply:
Added `todo!()` for those types that not obvious with `default` value.